new module: sale_team_default_values

lepistone opened the pull request.

lepistone committed:
move sale_team_default_values to OCA

lepistone committed:
set a default analytic account in the team

This functionality was previously in a separate, non-OCA module
analytic_sale_team.

elicoidal commented on sale_team_default_values/README.rst:
```diff
@@ -0,0 +1,4 @@
+Sales Teams Default Values
```
Hi @lepistone thanks for your contribution.

Please follow the OCA README template:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

elicoidal commented on sale_team_default_values/__init__.py:
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
```
Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

elicoidal commented on sale_team_default_values/__openerp__.py:
```diff
@@ -0,0 +1,35 @@
+##############################################################################
+
+{'name': 'Sales Teams Default Values',
+ 'version': '1.0',
```
version 8.0.1.0.0

elicoidal commented on sale_team_default_values/__openerp__.py:
```diff
@@ -0,0 +1,35 @@
+{'name': 'Sales Teams Default Values',
+ 'version': '1.0',
+ 'author': 'Camptocamp',
+ 'maintainer': 'Camptocamp',
```
remove key

elicoidal commented on sale_team_default_values/__openerp__.py:
```diff
@@ -0,0 +1,35 @@
+
+{'name': 'Sales Teams Default Values',
+ 'version': '1.0',
+ 'author': 'Camptocamp',
```
add OCA

elicoidal commented on sale_team_default_values/__openerp__.py:
```diff
@@ -0,0 +1,35 @@
+ 'author': 'Camptocamp',
+ 'maintainer': 'Camptocamp',
+ 'license': 'AGPL-3',
+ 'category': 'Specific',
```
Sales

elicoidal commented on sale_team_default_values/model/crm_case_section.py:
```diff
@@ -0,0 +1,45 @@
+#
+##############################################################################
+
+from openerp import models, fields
```
Alphabetical order

lepistone replied:
Where does this come from?

lepistone committed:
invoice: onchange team -> journal

lepistone committed:
propagate journal to generated invoice

lepistone committed:
readme, manifest, headers, authors

lepistone committed:
propagate only defined fields, rename them

A few fields are renamed so that they have the same name as on the sale
order.

lepistone committed:
fix field names

lepistone committed:
depend on sale_stock

lepistone committed:
fix field name project_id

lepistone committed:
fix: write an id, not a record

lepistone committed:
refactor method name

lepistone committed:
add test: propagate project

lepistone committed:
fix: do not remove default journal (with test)

lepistone committed:
test propagation of fiscal position

lepistone committed:
test pricelist propagation

lepistone committed:
fix header

lepistone committed:
use setattr: write does not work in onchanges

guewen commented on sale_team_default_values/model/sale_order.py:
```diff
@@ -0,0 +1,35 @@
+# Copyright 2014-2015 Camptocamp SA
+
+from openerp import models, api
+PROPAGATE_FIELDS = [
```
Using a method that returns the list of fields is better for extension, because with a globals such as this one, if a model does a `PROPAGATE_FIELDS.append('lambda_field')` in an extension addon, it could be executed even if the extension addon is not installed.

lepistone replied:
Agreed, done!

lepistone committed:
put list of fields in overridable method

lepistone committed:
tests: copy existing pricelist, so it has versions

This caused tests to fail when installed together with
sale_payment_term_interest.

lepistone committed:
make test compatible with sale_exception

rafaelbn commented:
Hi @lepistone , thanks a lot for this contribution.

It has been difficult to me understanding this module.

Please as @elicoidal said here https://github.com/OCA/sale-workflow/pull/235/files#r45451622 follow the OCA template, still missing:
- Installation
  - Just install module
- Configuration
  - No configuration needed
- Usage
  - Go to settings -> sales - sales teams and enable "Manage sales teams"
  - Go To Sales -> Sales teams -> Edit a sale team -> Go to second tab "default values" and punt the values you need for that team

This description is not valid because maybe the user who is going to review or use it never used another version before ;-)

```
Before Odoo 8.0, the sale.shop object contained some defaults for the sale process that were propagated to sale orders and invoices.

The shop was removed in 8.0, so this module uses the Sale Team as a replacement. Now a new tab in the Sales Team contains defaults that will be propagated to orders and invoices.
```

lepistone committed:
improve documentation

lepistone commented:
@rafaelbn Thanks for the review. I expanded the doc with `Configuration` and `Usage` sections, however I couldn't bring myself to write "Install: to install, install".

rousseldenis commented:
Closing this as no activity. @lepistone  Please feel free to reopen it if you still need it
